source/client.c

```diff
@@ -1460,6 +1469,8 @@ int aws_mqtt_client_connection_disconnect(
 
     { /* BEGIN CRITICAL SECTION */
         mqtt_connection_lock_synced_data(connection);
```

Comment:
an idea, maybe bad:
have this be macro so you can use `__FUNCTION__` when printing and see where it's being called from

Reply:
I chose to remove all the trace log instead😬